File: backend/src/main/java/com/vehiclerental/vehiclerental/service/VehicleService.java
package com.vehiclerental.vehiclerental.service;

import com.vehiclerental.vehiclerental.entity.Vehicle;
import java.util.List;

public interface VehicleService {
    List<Vehicle> getAllVehicles();
    Vehicle getVehicleById(Long id);
    Vehicle addVehicle(Vehicle vehicle);
    Vehicle updateVehicle(Long id, Vehicle vehicle);
    void deleteVehicle(Long id);
    List<Vehicle> filterVehicles(String type, String status);
}

File: frontend/src/services/vehicleService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/vehicles';

const getAllVehicles = (params) => {
    return axios.get(API_URL, { params });
};

const getVehicleById = (id) => {
    return axios.get(`${API_URL}/${id}`);
};

const addVehicle = (vehicle) => {
    return axios.post(API_URL, vehicle);
};

const updateVehicle = (id, vehicle) => {
    return axios.put(`${API_URL}/${id}`, vehicle);
};

const deleteVehicle = (id) => {
    return axios.delete(`${API_URL}/${id}`);
};

const vehicleService = {
    getAllVehicles,
    getVehicleById,
    addVehicle,
    updateVehicle,
    deleteVehicle
};

export default vehicleService;
